src/librustc_resolve/lib.rs

```diff
@@ -3517,10 +3517,10 @@ fn path_names_to_string(path: &Path, depth: usize) -> String {
 }
 
 /// A somewhat inefficient routine to obtain the name of a module.
-fn module_to_string<'a>(module: Module<'a>) -> String {
+fn module_to_string(module: Module) -> String {
     let mut names = Vec::new();
 
-    fn collect_mod<'a>(names: &mut Vec<ast::Name>, module: Module<'a>) {
```

Comment:
Personally, I don't find the code to be any more clear with the explicit lifetimes. What's the concern here? I don't feel strongly, just curious